element_calcium_imaging/imaging.py

```diff
@@ -119,6 +120,29 @@ class ProcessingTask(dj.Manual):
     task_mode='load': enum('load', 'trigger')   # 'load': load computed analysis results, 'trigger': trigger computation
     """
 
+    @classmethod
+    def auto_generate_entries(cls, scan_key, task_mode):
+        """
+        Method to auto-generate ProcessingTask entries for a particular Scan.
+        """
+
+        image_locators = [get_scan_image_files, get_scan_box_files, get_nd2_files]
+        for image_locator in image_locators:
+            try:
+                relative_scan_dir = pathlib.Path(image_locator(scan_key)[0]).relative_to(get_imaging_root_data_dir()).parent
+            except:
+                pass
+            if relative_scan_dir:
+                break
+        
+        paramset_idxs = ProcessingParamSet.fetch('KEY')
+
+        for i, paramset_idx in enumerate(paramset_idxs):
+            processing_method = paramset_idx['processing_method'][i]
+            processing_output_dir = (get_processed_root_data_dir() / relative_scan_dir / f'{processing_method}_{paramset_idx}').as_posix()
+
+            cls.insert1({**scan_key, 'paramset_idx':paramset_idx, 'processing_output_dir': processing_output_dir, 'task_mode': task_mode})
```

Comment:
Let's not loop through all paramset_idx either, we should have only one paramset_idx to autogenerate `processing_task` from, a default one.